pkg/kubectl/generate_test.go

```diff
@@ -138,3 +138,243 @@ func TestMakeParams(t *testing.T) {
 		t.Errorf("\nexpected:\n%v\nsaw:\n%v", expected, params)
 	}
 }
+
+func TestGetBool(t *testing.T) {
+	testCases := []struct {
+		name         string
+		parameters   map[string]string
+		key          string
+		defaultValue bool
+		expected     bool
+		expectError  bool
+	}{
+		{
+			name: "found key in parameters, default value is different from key value",
+			parameters: map[string]string{
+				"foo": "false",
+			},
+			key:          "foo",
+			defaultValue: false,
+			expected:     false,
+			expectError:  false,
+		},
+		{
+			name: "found key in parameters, default value is same with key value",
+			parameters: map[string]string{
+				"foo": "true",
+			},
+			key:          "foo",
+			defaultValue: true,
+			expected:     true,
+			expectError:  false,
+		},
+		{
+			name: "key not found in parameters, default value is true",
+			parameters: map[string]string{
+				"foo": "true",
+				"far": "false",
+			},
+			key:          "bar",
+			defaultValue: true,
+			expected:     true,
+			expectError:  false,
+		},
+		{
+			name: "key not found in parameters, default value is false",
+			parameters: map[string]string{
+				"foo": "true",
+				"far": "false",
+			},
+			key:          "bar",
+			defaultValue: false,
+			expected:     false,
+			expectError:  false,
+		},
+		{
+			name:         "parameters is empty",
+			parameters:   map[string]string{},
+			key:          "foo",
+			defaultValue: true,
+			expected:     true,
+			expectError:  false,
+		},
+		{
+			name: "parameters key is not a valid bool value",
+			parameters: map[string]string{
+				"foo": "error",
+			},
+			key:          "foo",
+			defaultValue: true,
+			expected:     false,
+			expectError:  true,
+		},
+	}
+	for _, test := range testCases {
+		got, err := GetBool(test.parameters, test.key, test.defaultValue)
+		if err != nil && test.expectError == false {
+			t.Errorf("%s: unexpected error: %v", test.name, err)
+		}
+		if err == nil && test.expectError == true {
+			t.Errorf("%s: expect error, got nil", test.name)
+		}
+		if got != test.expected {
+			t.Errorf("%s: expect %v, got %v", test.name, test.expected, got)
+		}
+	}
+}
+
+func TestMakeParseLabels(t *testing.T) {
+	successCases := []struct {
```

Comment:
In successCases, I use function `MakeLabels()` to generate labelString, so the test cases covered function `MakeLabels()` as well. Since function `MakeLabels()` will always generate valid labelString, so I construct the labelString by hand in errorCases. That's why I created successCases and errorCases.

Please let me know if you have better idea.